weave-js/src/components/PagePanelComponents/persistenceStateMachine.ts

```diff
@@ -272,15 +272,22 @@ export const useStateMachine = (
             artifact_name:
               actionOptions.name != null
                 ? constString(toArtifactSafeName(actionOptions.name))
-                : constNone(), // TODO: Allow user to specify name
-            project_name: constString('weave'), // TODO: Allow user to specify name
+                : constNone(),
+            project_name:
+              actionOptions.projectName != null
+                ? constString(actionOptions.projectName)
+                : constString('weave'),
+            entity_name:
+              actionOptions.entityName != null
+                ? constString(actionOptions.entityName)
+                : constString(''),
```

Comment:
should the default here be a constNone()?

Reply:
Probably yes - I think in the current code you can't actually hit this path